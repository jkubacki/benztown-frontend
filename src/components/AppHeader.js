import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';

import LogoutButton from 'components/LogoutButton'
import SearchForm from 'components/SearchForm'

import ClientState from 'constants/clientState'

import './AppHeader.css';

function AppHeader({ isLoggedIn }) {
  return (
    <div className="navbar">
      <ul className="navbar-nav nav-fill d-flex w-100 flex-md-row">
        <li className="nav-item">
          <span className="navbar-brand">Benztown</span>
        </li>
        {
          isLoggedIn &&
          <li className="nav-item">
            <SearchForm />
          </li>
        }
        {
          isLoggedIn &&
          <li className="nav-item">
            <LogoutButton />
          </li>
        }
      </ul>
    </div>
  )
}

export default compose(
  connect(({ client }) => ({
    isLoggedIn: client.state === ClientState.LOGGED_IN,
  })),
)(AppHeader);
