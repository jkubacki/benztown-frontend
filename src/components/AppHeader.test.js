import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AppHeader } from './AppHeader';

const props = { isLoggedIn: false };

it('renders without crashing when logged in', () => {
  const component = shallow(<AppHeader {...props} isLoggedIn />);
  expect(toJson(component)).toMatchSnapshot();
});

it('renders without crashing when not logged in', () => {
  const component = shallow(<AppHeader {...props} />);
  expect(toJson(component)).toMatchSnapshot();
});
