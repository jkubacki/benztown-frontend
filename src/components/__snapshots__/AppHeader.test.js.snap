// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing when logged in 1`] = `
<div
  className="navbar"
>
  <ul
    className="navbar-nav nav-fill d-flex w-100 flex-md-row"
  >
    <li
      className="nav-item"
    >
      <span
        className="navbar-brand"
      >
        Benztowny
      </span>
    </li>
    <li
      className="nav-item"
    >
      <Connect(SearchForm) />
    </li>
    <li
      className="nav-item"
    >
      <Connect(LogoutButton) />
    </li>
  </ul>
</div>
`;

exports[`renders without crashing when not logged in 1`] = `
<div
  className="navbar"
>
  <ul
    className="navbar-nav nav-fill d-flex w-100 flex-md-row"
  >
    <li
      className="nav-item"
    >
      <span
        className="navbar-brand"
      >
        Benztowny
      </span>
    </li>
  </ul>
</div>
`;
